lib/honeycomb/trace.rb

```diff
@@ -34,5 +34,17 @@ def initialize(builder:, context:, serialized_trace: nil, **options)
     def add_field(key, value)
       @fields[key] = value
     end
+
+    private
+    
+    INVALID_TRACE_ID = ("\0" * 16).b
```

Comment:
Thinking on this further, is this ever going to match something generated by `SecureRandom.hex`?

Reply:
Nope! But see comment below, I dug into the bytes discrepancy briefly and we should move this over to raw bytes anyway. Also it's kind of interesting that ruby otel does this comparison on id generation. More commonly seen, is that a parser called on an incoming header will check against this invalid trace or span id and return undefined if it's detected.

Reply:
surely you'd have to do the comparison here or you could generate spans/traces with invalid ids that would only be discovered when trying to send them?

Reply:
yeah that's a good point. I should probably add this check to nodejs